pandas/tests/indexes/categorical/test_category.py

```diff
@@ -286,6 +286,17 @@ def test_map_str(self):
         # See test_map.py
         pass
 
+    def test_append(self):
+        # GH 44099
+        # concat indexes which have the same categories
+
+        ci1 = CategoricalIndex(["a", "b", "c"], categories=["a", "b", "c"])
+        ci2 = CategoricalIndex(["b", "a", "c"], categories=["b", "a", "c"])
+        expected = CategoricalIndex(
+            ["a", "b", "c", "b", "a", "c"], categories=["a", "b", "c"]
+        )
+        tm.assert_index_equal(ci1.append(ci2), expected)
```

Comment:
use ```result=ci1.append(ci2)```

Reply:
I also changed at this commit https://github.com/pandas-dev/pandas/commit/91d215568867985d1014f680d60d41f61f33af1e